### PR TITLE
Disable exceptions in zstr

### PR DESCRIPTION
**Pull request by jandrej**
Fix for #1412

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1502](https://github.com/mfem/mfem/pull/1502) | @jandrej | @tzanio | @acfisher + @v-dobrev | 05/25/20 | 06/01/20 | 06/07/20 | |

### CODE REVIEW

**Comment by tzanio** [comment]
@acfisher, any chance you can take a quick look (this bugfix is relatively minor)?

File: general/zstr.hpp [hunk 1]
```diff
@@ -203,10 +203,10 @@ class ifstream
    {
       mode |= std::ios_base::in;
       exceptions(std::ios_base::badbit);
-      detail::static_method_holder::check_mode(filename, mode);
+      // detail::static_method_holder::check_mode(filename, mode);
```
Comment (acfisher) [inline general/zstr.hpp:206]: Is the plan to fix these checks later or should we just remove these?

Comment (jandrej) [reply general/zstr.hpp:206]: My intention of just commenting the lines was to keep the "diff" to the original zstr source. The fix for the exception is this PR. Regular behavior of an iostream is to only set a failbit.

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**